### DISCUSSION TITLE
Dropwizard 2.1.0 compatible release

### DISCUSSION THREAD

**Discussion by xvik** [xvik/dropwizard-guicey#226] [General]
For information. 

Dropwizard 2.1.0 was released, but there is [no release notes yet](https://github.com/dropwizard/dropwizard/discussions/5283), only [upgrade guide](https://www.dropwizard.io/en/latest/manual/upgrade-notes/upgrade-notes-2_1_x.html).

Guicey already fixed for 2.1.0:
1. There was a problem in configuration values binding (you might not be affected)
2. GSP and SPA ext modules were affected by change in asset servlet behaviour (for 404 response)

Known issues:
* There is a nasty [warning (stackthace) under java 8](https://github.com/dropwizard/dropwizard/discussions/5268#discussioncomment-2723607) now which can't be easilly avoided. Everything works, but looks scary.
* [Httpclient problem](https://github.com/dropwizard/dropwizard/issues/5281). Probably soon will be a bugfix release.

I'm currently **waiting with new guicey release**, especially because of java 8 warning (it could be avoided only by manual ObjectMapper construction for Bootsrtap, but this would mean manual synchronization of [registered modules](https://github.com/dropwizard/dropwizard/blob/release/2.1.x/dropwizard-jackson/src/main/java/io/dropwizard/jackson/Jackson.java#L58) with new dropwizard versions).

I fixed [snapshot builds on jetpack](https://github.com/xvik/dropwizard-guicey#snapshots), so you can already try master-SNAPSHOT guicey version (from jitpack repo).

Please write here if you need urgent release or have any thoughts on current issues.

**Comment by ron-ak-p** [comment]
Since [Dropwizard 2.1 GA](https://github.com/dropwizard/dropwizard/releases/tag/v2.1.0) was released early last week, we are looking to upgrade our services to it ASAP, primarily to address [CVE-2020-36518](https://github.com/advisories/GHSA-57j2-w4cx-62h2). We use dropwizard-guicey in all of those services, so we would appreciate a DW 2.1 compatible release soon. Thank you!

**Reply by xvik** [reply]
Will release it this weekend. 
I was hoping they would release a hotfix version fixing current java 8 issue, but [discussion lead to nowhere currently](https://github.com/dropwizard/dropwizard/discussions/5319)

**Reply by ron-ak-p** [reply]
Thank you!
That's unfortunate. This "issue" is going to affect our services, since they're all still on Java 8.

**Comment by xvik** [comment]
Guicey 5.6.0 released. See [release notes](http://xvik.github.io/dropwizard-guicey/latest/about/release-notes/) for details (especially for java 8 issue)

**Comment by xvik** [comment]
About java 8 problem: If there anyone want it to be resolved please [look the last messages in dropwizard discussion](https://github.com/dropwizard/dropwizard/discussions/5319). I'm out of arguments. If you have anything to say about it, please say it.

Summary: 

A new simpler workaround was proposed:
```java
((LoggerContext)org.slf4j.LoggerFactory.getILoggerFactory()).getLogger(BlackbirdModule.class).setLevel(Level.ERROR);
```
and it works!

And there was a [ObjectMapperFactory proposition](https://github.com/dropwizard/dropwizard/pull/5472) which would simplify ObjectMapper customization, but still will not remove the warning by default.

Proposition to simply not register blackbird for java 8 was not accepted.

**Comment by xvik** [comment]
Dropwizard 2.1.1 released with a java 8 issue fix: it is now enough to somply add afterberner to classpath and it would be used instead of blackbird automatically.

New [guicey 5.6.1 released](https://github.com/xvik/dropwizard-guicey/releases/tag/5.6.1)
[Release notes](http://xvik.github.io/dropwizard-guicey/latest/about/release-notes/) were also slightly updated
